app/controllers/api/account.php

```diff
@@ -1677,6 +1677,95 @@
+    ->param('providerId', '', new UID(), 'Provider ID. Message will be sent to this target from the specified provider ID. If no provider ID is set the first setup provider will be used.')
+    ->param('identifier', '', new Text(Database::LENGTH_KEY), 'The target identifier (token, email, phone etc.)')
+    ->inject('queueForEvents')
+    ->inject('user')
+    ->inject('request')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $targetId, string $providerId, string $identifier, Event $queueForEvents, Document $user, Request $request, Response $response, Database $dbForProject) {
+        $targetId = $targetId == 'unique()' ? ID::unique() : $targetId;
+
+        $provider = Authorization::skip(fn () => $dbForProject->getDocument('providers', $providerId));
+
+        if ($provider->isEmpty()) {
+            throw new Exception(Exception::PROVIDER_NOT_FOUND);
+        }
+
+        if ($user->isEmpty()) {
+            throw new Exception(Exception::USER_NOT_FOUND);
+        }
+
+        $target = Authorization::skip(fn () => $dbForProject->getDocument('targets', $targetId));
+
+        if (!$target->isEmpty()) {
+            throw new Exception(Exception::USER_TARGET_ALREADY_EXISTS);
+        }
+
+        $detector = new Detector($request->getUserAgent());
+        $detector->skipBotDetection(); // OPTIONAL: If called, bot detection will completely be skipped (bots will be detected as regular devices then)
+
+        $os = $detector->getOS();
+        $client = $detector->getClient();
+        $device = $detector->getDevice();
+
+        try {
+            $target = $dbForProject->createDocument('targets', new Document([
+                '$id' => $targetId,
+                '$permissions' => [
+                    Permission::read(Role::user($user->getId())),
+                    Permission::update(Role::user($user->getId())),
+                ],
+                'providerId' => $providerId ?? null,
+                'providerInternalId' => $provider->getInternalId() ?? null,
+                'providerType' =>  'push',
+                'userId' => $user->getId(),
+                'userInternalId' => $user->getInternalId(),
+                'identifier' => $identifier,
+                'name' => [
+                    'osCode' => $os['osCode'],
+                    'osName' => $os['osName'],
+                    'osVersion' => $os['osVersion'],
+                    'clientType' => $client['clientType'],
+                    'clientCode' => $client['clientCode'],
+                    'clientName' => $client['clientName'],
+                    'clientVersion' => $client['clientVersion'],
+                    'clientEngine' => $client['clientEngine'],
+                    'clientEngineVersion' => $client['clientEngineVersion'],
+                    'deviceBrand' => $device['deviceBrand'],
+                    'deviceModel' => $device['deviceModel']
+                ]
```

Comment:
Let's just shortcut here:

```suggestion
                'name' => "{$device['deviceBrand']} {$device['deviceModel']}"
```

Reply:
done this